System.server crash

```
type: crash
flags: dev options enabled

SystemUptimeMs: 315008727
Process: system_server
Timestamp: 2024-11-28 08:51:22.121+0100
Build: google/bluejay/bluejay:15/AP3A.241105.007/2024111800:user/release-keys
Crash-Handler: com.android.internal.os.RuntimeInit$KillApplicationHandler
Loading-Progress: 1.0
Dropped-Count: 0

java.lang.AssertionError: java.lang.IllegalArgumentException: provider "network" does not exist
	at android.location.provider.LocationProviderBase$Service.lambda$setRequest$0(LocationProviderBase.java:336)
	at android.location.provider.LocationProviderBase$Service$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:959)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at com.android.server.SystemServer.run(SystemServer.java:983)
	at com.android.server.SystemServer.main(SystemServer.java:661)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:880)
Caused by: java.lang.IllegalArgumentException: provider "network" does not exist
	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:64)
	at com.android.server.location.LocationManagerService.registerLocationListener(LocationManagerService.java:845)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1618)
	at com.android.location.fused.FusedLocationProvider$ChildLocationListener.resetProviderRequest(FusedLocationProvider.java:304)
	at com.android.location.fused.FusedLocationProvider$ChildLocationListener.-$$Nest$mresetProviderRequest(FusedLocationProvider.java:0)
	at com.android.location.fused.FusedLocationProvider.updateRequirementsLocked(FusedLocationProvider.java:173)
	at com.android.location.fused.FusedLocationProvider.onSetRequest(FusedLocationProvider.java:123)
	at android.location.provider.LocationProviderBase$Service.setRequest(LocationProviderBase.java:331)
	at com.android.server.location.provider.proxy.ProxyLocationProvider.lambda$onSetRequest$0(ProxyLocationProvider.java:184)
	at com.android.server.location.provider.proxy.ProxyLocationProvider.$r8$lambda$7VMdFID5UKJaFx-7qHYncL9kLJk(ProxyLocationProvider.java:0)
	at com.android.server.location.provider.proxy.ProxyLocationProvider$$ExternalSyntheticLambda1.run(R8$$SyntheticClass:0)
	at com.android.server.servicewatcher.ServiceWatcherImpl$MyServiceConnection.runOnBinder(ServiceWatcherImpl.java:257)
	at com.android.server.servicewatcher.ServiceWatcherImpl.lambda$runOnBinder$0(ServiceWatcherImpl.java:126)
	at com.android.server.servicewatcher.ServiceWatcherImpl.$r8$lambda$GCGDYqQZRK7jCQUNZnY8QjXgR2Y(ServiceWatcherImpl.java:0)
	at com.android.server.servicewatcher.ServiceWatcherImpl$$ExternalSyntheticLambda1.run(R8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:959)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.os.HandlerThread.run(HandlerThread.java:85)
	at com.android.server.ServiceThread.run(ServiceThread.java:46)
```

```
type: crash
flags: dev options enabled

SystemUptimeMs: 116120713
Process: system_server
Timestamp: 2024-12-03 11:44:36.800+0000
Build: google/raven/raven:15/AP3A.241105.007/2024112700:user/release-keys
Crash-Handler: com.android.internal.os.RuntimeInit$KillApplicationHandler
Loading-Progress: 1.0
Dropped-Count: 0

java.lang.AssertionError: java.lang.IllegalArgumentException: provider "network" does not exist
	at android.location.provider.LocationProviderBase$Service.lambda$setRequest$0(LocationProviderBase.java:336)
	at android.location.provider.LocationProviderBase$Service$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:959)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at com.android.server.SystemServer.run(SystemServer.java:983)
	at com.android.server.SystemServer.main(SystemServer.java:661)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:880)
Caused by: java.lang.IllegalArgumentException: provider "network" does not exist
	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:64)
	at com.android.server.location.LocationManagerService.registerLocationListener(LocationManagerService.java:845)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1618)
	at com.android.location.fused.FusedLocationProvider$ChildLocationListener.resetProviderRequest(FusedLocationProvider.java:304)
	at com.android.location.fused.FusedLocationProvider$ChildLocationListener.-$$Nest$mresetProviderRequest(FusedLocationProvider.java:0)
	at com.android.location.fused.FusedLocationProvider.updateRequirementsLocked(FusedLocationProvider.java:173)
	at com.android.location.fused.FusedLocationProvider.onSetRequest(FusedLocationProvider.java:123)
	at android.location.provider.LocationProviderBase$Service.setRequest(LocationProviderBase.java:331)
	at com.android.server.location.provider.proxy.ProxyLocationProvider.lambda$onSetRequest$0(ProxyLocationProvider.java:184)
	at com.android.server.location.provider.proxy.ProxyLocationProvider.$r8$lambda$7VMdFID5UKJaFx-7qHYncL9kLJk(ProxyLocationProvider.java:0)
	at com.android.server.location.provider.proxy.ProxyLocationProvider$$ExternalSyntheticLambda1.run(R8$$SyntheticClass:0)
	at com.android.server.servicewatcher.ServiceWatcherImpl$MyServiceConnection.runOnBinder(ServiceWatcherImpl.java:257)
	at com.android.server.servicewatcher.ServiceWatcherImpl.lambda$runOnBinder$0(ServiceWatcherImpl.java:126)
	at com.android.server.servicewatcher.ServiceWatcherImpl.$r8$lambda$GCGDYqQZRK7jCQUNZnY8QjXgR2Y(ServiceWatcherImpl.java:0)
	at com.android.server.servicewatcher.ServiceWatcherImpl$$ExternalSyntheticLambda1.run(R8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:959)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.os.HandlerThread.run(HandlerThread.java:85)
	at com.android.server.ServiceThread.run(ServiceThread.java:46)
```

i use ivpn app for mock gps, seems to crash randomly, the times ive noticed it happen the vpn is not enabled

--- a/packages/FusedLocation/src/com/android/location/fused/FusedLocationService.java
+++ b/packages/FusedLocation/src/com/android/location/fused/FusedLocationService.java
@@ -26,30 +26,27 @@ import java.io.PrintWriter;
 
 public class FusedLocationService extends Service {
 
-    @Nullable private FusedLocationProvider mProvider;
+    private FusedLocationProvider mProvider;
+
+    @Override
+    public void onCreate() {
+        mProvider = new FusedLocationProvider(this);
+        mProvider.start();
+    }
 
     @Override
     public IBinder onBind(Intent intent) {
-        if (mProvider == null) {
-            mProvider = new FusedLocationProvider(this);
-            mProvider.start();
-        }
-
         return mProvider.getBinder();
     }
 
     @Override
     public void onDestroy() {
-        if (mProvider != null) {
-            mProvider.stop();
-            mProvider = null;
-        }
+        mProvider.stop();
+        mProvider = null;
     }
 
     @Override
     protected void dump(FileDescriptor fd, PrintWriter writer, String[] args) {
-        if (mProvider != null) {
-            mProvider.dump(writer);
-        }
+        mProvider.dump(writer);
     }
 }

--- a/packages/FusedLocation/src/com/android/location/gnss/GnssOverlayLocationService.java
+++ b/packages/FusedLocation/src/com/android/location/gnss/GnssOverlayLocationService.java
@@ -26,24 +26,23 @@ import java.io.PrintWriter;
 
 public class GnssOverlayLocationService extends Service {
 
-    @Nullable private GnssOverlayLocationProvider mProvider;
+    private GnssOverlayLocationProvider mProvider;
+
+    @Override
+    public void onCreate() {
+        mProvider = new GnssOverlayLocationProvider(this);
+        mProvider.start();
+    }
 
     @Override
     public IBinder onBind(Intent intent) {
-        if (mProvider == null) {
-            mProvider = new GnssOverlayLocationProvider(this);
-            mProvider.start();
-        }
-
         return mProvider.getBinder();
     }
 
     @Override
     public void onDestroy() {
-        if (mProvider != null) {
-            mProvider.stop();
-            mProvider = null;
-        }
+        mProvider.stop();
+        mProvider = null;
     }
 
     @Override

--- a/services/core/java/com/android/server/location/LocationManagerService.java
+++ b/services/core/java/com/android/server/location/LocationManagerService.java
@@ -392,16 +392,6 @@ public class LocationManagerService extends ILocationManager.Stub implements
         }
     }
 
-    private void removeLocationProviderManager(LocationProviderManager manager) {
-        synchronized (mProviderManagers) {
-            boolean removed = mProviderManagers.remove(manager);
-            Preconditions.checkArgument(removed);
-            manager.setMockProvider(null);
-            manager.setRealProvider(null);
-            manager.stopManager();
-        }
-    }
-
     void onSystemReady() {
         if (Build.IS_DEBUGGABLE) {
             // on debug builds, watch for location noteOps while location is off. there are some
@@ -1493,9 +1483,6 @@ public class LocationManagerService extends ILocationManager.Stub implements
             }
 
             manager.setMockProvider(null);
-            if (!manager.hasProvider()) {
-                removeLocationProviderManager(manager);
-            }
         }
     }
 

--- a/services/core/java/com/android/server/location/provider/LocationProviderManager.java
+++ b/services/core/java/com/android/server/location/provider/LocationProviderManager.java
@@ -1589,36 +1589,6 @@ public class LocationProviderManager extends
         }
     }
 
-    public void stopManager() {
-        synchronized (mMultiplexerLock) {
-            Preconditions.checkState(mState == STATE_STARTED);
-            mState = STATE_STOPPING;
-
-            final long identity = Binder.clearCallingIdentity();
-            try {
-                onEnabledChanged(UserHandle.USER_ALL);
-                removeRegistrationIf(key -> true);
-                mProvider.getController().stop();
-            } finally {
-                Binder.restoreCallingIdentity(identity);
-            }
-
-            mUserHelper.removeListener(mUserChangedListener);
-            mLocationSettings.unregisterLocationUserSettingsListener(mLocationUserSettingsListener);
-            mSettingsHelper.removeOnLocationEnabledChangedListener(mLocationEnabledChangedListener);
-
-            // if external entities are registering listeners it's their responsibility to
-            // unregister them before stopManager() is called
-            Preconditions.checkState(mEnabledListeners.isEmpty());
-            mProviderRequestListeners.clear();
-
-            mEnabled.clear();
-            mLastLocations.clear();
-            mStateChangedListener = null;
-            mState = STATE_STOPPED;
-        }
-    }
-
     public String getName() {
         return mName;
     }

